.github/workflows/packaging.yml

```diff
@@ -120,11 +120,62 @@ jobs:
           path: |
             apache-arrow-adbc-${{ steps.version.outputs.VERSION }}.tar.gz
 
+  go-binaries:
+    name: "Go ${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    needs:
+      - source
+
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "windows-latest", "macos-13"]
+        go-version: [1.21.8] # Customize Go versions as needed
+
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: source
+
+      - name: Extract source archive
+        run: |
+          source_archive=$(echo apache-arrow-adbc-*.tar.gz)
+          VERSION=${source_archive#apache-arrow-adbc-}
+          VERSION=${VERSION%.tar.gz}
+          echo "VERSION=${VERSION}" >> $GITHUB_ENV
+
+          tar xf apache-arrow-adbc-${VERSION}.tar.gz
+          mv apache-arrow-adbc-${VERSION} adbc
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
+          check-latest: true
+          cache: true
+          cache-dependency-path: adbc/go/adbc/go.sum
+
+      # run `make -i` because the Windows runs claim they can't delete the .h files, but they are still generated
+      - name: Build Go binaries
+        run: |
+          pushd adbc/go/adbc/pkg
+          make -i
+          popd
+
+      - name: Upload Go binaries
+        uses: actions/upload-artifact@v3
```

Comment:
```suggestion
        uses: actions/upload-artifact@v4
```

Reply:
I changed them to v4 but it was breaking and not uploading the files correctly, so I went back to v3.